### PR TITLE
dev-db/mongodb: updates

### PR DESCRIPTION
**Pull request by Dessa**
first of all im very open to just remove 4.4 at this point, people had enough time to upgrade, but i still provided it and since its py3k12 compliant now it should not be a blocker.
now having said that the removal of the enterprise modules (!) kind of doubles as a python 3.12 fix so while its the easy way out, i also believe its the right thing, despite other distros seemingly (?) including them. their license agreement for that part seems EULA'y so i, to be perfectly honest, didn't wanna deal with it.
i have masked it on musl, there is really no hope of this ever working there (as of what will be, 8.0, there is still no effort being made towards that)

```
--- mongodb-5.0.16-r1.ebuild	2024-02-18 22:25:37.334245855 +0100
+++ mongodb-5.0.26.ebuild	2024-05-06 22:05:48.467982656 +0200
@@ -1,26 +1,28 @@
 # Copyright 1999-2024 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=7
+EAPI=8
 
-PYTHON_COMPAT=( python3_{10..11} )
+PYTHON_COMPAT=( python3_{10..12} )
 
 SCONS_MIN_VERSION="3.3.1"
 CHECKREQS_DISK_BUILD="2400M"
 CHECKREQS_DISK_USR="512M"
 CHECKREQS_MEMORY="1024M"
 
-inherit check-reqs flag-o-matic multiprocessing pax-utils python-any-r1 scons-utils systemd toolchain-funcs
+inherit check-reqs flag-o-matic multiprocessing pax-utils python-any-r1 scons-utils systemd toolchain-funcs verify-sig
 
 MY_P=${PN}-src-r${PV/_rc/-rc}
 
 DESCRIPTION="A high-performance, open source, schema-free document-oriented database"
 HOMEPAGE="https://www.mongodb.com"
 SRC_URI="https://fastdl.mongodb.org/src/${MY_P}.tar.gz"
+SRC_URI+=" verify-sig? ( https://fastdl.mongodb.org/src/${MY_P}.tar.gz.sig )"
+S="${WORKDIR}/${MY_P}"
 
 LICENSE="Apache-2.0 SSPL-1"
 SLOT="0"
-KEYWORDS="amd64 ~arm64 -riscv"
+KEYWORDS="~amd64 ~arm64 -riscv"
 CPU_FLAGS="cpu_flags_x86_avx"
 IUSE="debug kerberos mongosh ssl +tools ${CPU_FLAGS}"
 
@@ -30,14 +32,13 @@
 
 RDEPEND="acct-group/mongodb
 	acct-user/mongodb
-	>=app-arch/snappy-1.1.3:=
+	>=app-arch/snappy-1.1.7:=
 	>=dev-cpp/yaml-cpp-0.6.2:=
 	dev-libs/boost:=[nls]
 	>=dev-libs/libpcre-8.42[cxx]
 	app-arch/zstd:=
 	dev-libs/snowball-stemmer:=
-	net-libs/libpcap
-	>=sys-libs/zlib-1.2.11:=
+	>=sys-libs/zlib-1.2.12:=
 	kerberos? ( dev-libs/cyrus-sasl[kerberos] )
 	ssl? (
 		>=dev-libs/openssl-1.0.1g:0=
@@ -54,12 +55,15 @@
 		dev-python/psutil[${PYTHON_USEDEP}]
 		dev-python/pyyaml[${PYTHON_USEDEP}]
 	')
+	verify-sig? ( ~sec-keys/openpgp-keys-mongodb-$(ver_cut 1-2) )
 "
 PDEPEND="
 	mongosh? ( app-admin/mongosh-bin )
 	tools? ( >=app-admin/mongo-tools-100 )
 "
 
+VERIFY_SIG_OPENPGP_KEY_PATH="/usr/share/openpgp-keys/mongodb-$(ver_cut 1-2).asc"
+
 PATCHES=(
 	"${FILESDIR}/${PN}-4.4.1-boost.patch"
 	"${FILESDIR}/${PN}-4.4.1-gcc11.patch"
@@ -71,17 +75,14 @@
 	"${FILESDIR}/${PN}-5.0.5-no-force-lld.patch"
 	"${FILESDIR}/${PN}-4.4.10-boost-1.81.patch"
 	"${FILESDIR}/${PN}-5.0.5-boost-1.81-extra.patch"
-	"${FILESDIR}/${PN}-4.4.8-gcc-13.patch"
-	"${FILESDIR}/${PN}-5.0.16-arm64-assert.patch"
+	"${FILESDIR}/${PN}-4.4.29-no-enterprise.patch"
 )
 
-S="${WORKDIR}/${MY_P}"
-
 python_check_deps() {
-	python_has_version ">=dev-build/scons-3.1.1[${PYTHON_USEDEP}]" &&
-	python_has_version "dev-python/cheetah3[${PYTHON_USEDEP}]" &&
-	python_has_version "dev-python/psutil[${PYTHON_USEDEP}]" &&
-	python_has_version "dev-python/pyyaml[${PYTHON_USEDEP}]"
+	python_has_version -b ">=dev-build/scons-3.1.1[${PYTHON_USEDEP}]" &&
+	python_has_version -b "dev-python/cheetah3[${PYTHON_USEDEP}]" &&
+	python_has_version -b "dev-python/psutil[${PYTHON_USEDEP}]" &&
+	python_has_version -b "dev-python/pyyaml[${PYTHON_USEDEP}]"
 }
 
 pkg_pretend() {
@@ -89,7 +90,7 @@
 	if use amd64 && ! use cpu_flags_x86_avx; then
 		ewarn "MongoDB 5.0 requires use of the AVX instruction set."
 		ewarn "This ebuild will use --experimental-optimization=-sandybridge which"
-	        ewarn "will result in an experimental build of MongoDB as per upstream."
+		ewarn "will result in an experimental build of MongoDB as per upstream."
 		ewarn "https://docs.mongodb.com/v5.0/administration/production-notes/"
 	fi
 
@@ -122,9 +123,13 @@
 		AR="$(tc-getAR)"
 		CC="$(tc-getCC)"
 		CXX="$(tc-getCXX)"
+		CCFLAGS="${CXXFLAGS}"
+
+		VERBOSE=1
+		VARIANT_DIR=gentoo
+		--modules=""
 
 		--disable-warnings-as-errors
-		--force-jobs
 		--jobs="$(makeopts_jobs)"
 		--use-system-boost
 		--use-system-pcre
@@ -136,9 +141,9 @@
 	)
 
 	use arm64 && scons_opts+=( --use-hardware-crc32=off ) # Bug 701300
+	use amd64 && scons_opts+=( --experimental-optimization=-sandybridge ) # Bug 890294
 	use debug && scons_opts+=( --dbg=on )
 	use kerberos && scons_opts+=( --use-sasl-client )
-	use amd64 && ! use cpu_flags_x86_avx && scons_opts+=( --experimental-optimization=-sandybridge ) # Bug 890294
 
 	scons_opts+=( --ssl=$(usex ssl on off) )
 
@@ -146,6 +151,13 @@
 	# Gentoo's toolchain applies these anyway
 	scons_opts+=( --runtime-hardening=off )
 
+	# gold is an option here but we don't really do that anymore
+	if tc-ld-is-lld; then
+		 scons_opts+=( --linker=lld )
+	else
+		 scons_opts+=( --linker=bfd )
+	fi
+
 	# respect mongoDB upstream's basic recommendations
 	# see bug #536688 and #526114
 	if ! use debug; then
@@ -183,11 +195,8 @@
 	# see bug #526114
 	pax-mark emr "${ED}"/usr/bin/{mongo,mongod,mongos}
 
-	local x
-	for x in /var/{lib,log}/${PN}; do
-		diropts -m0750 -o mongodb -g mongodb
-		keepdir "${x}"
-	done
+	diropts -m0750 -o mongodb -g mongodb
+	keepdir /var/log/${PN}
 }
 
 pkg_postinst() {

```

### CODE REVIEW

**Comment by ultrabug** [comment]
Thanks a lot for your work @Dessa 

I'll try and build that locally to validate everything and we'll be good to go.

PS: would you be interested in bumping mongodb major version in a separate PR?

**Comment by Dessa** [comment]
yes bumping to 6 timely after this hits the tree is the plan here.

not sure about 7 yet, given the package still follows the "upgrade the database to each major version first" approach, so i feel like covering 6 *and* 7 will end up in a bad user experience, but so would be just masking 7 for a while, im at a loss here how to handle it properly, given that it is also a package with not the shortest build time.

File: profiles/features/musl/package.mask [hunk 1]
```diff
@@ -1,6 +1,17 @@
 # Copyright 1999-2024 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
+# Robert Förster <Dessa@gmake.de> (2024-05-06)
+# needs mongodb
+media-video/unifi-video
```
Comment (ultrabug) [inline profiles/features/musl/package.mask:6]: this package was removed yesterday, can you drop it please?

**Review by ultrabug — CHANGES_REQUESTED** [review]
builds OK here, thanks for the work

**Comment by ultrabug** [comment]
Thanks a lot @Dessa !

**Comment by adlerweb** [comment]
I just noticed this commit removed --force-jobs, in my case this reintroduced https://bugs.gentoo.org/906897 (`ERROR: Icecream not enabled while using -j higher than available cpu's. Use --force-jobs to override.`).
@Dessa Was there any specific reason why this flag was removed?

**Pull request closed without merging**